cpu/native/netdev_raw802154/netdev_raw802154.c

```diff
@@ -0,0 +1,337 @@
+/*
+ * Copyright (C) 2016 Alexander Aring <aar@pengutronix.de>
+ * Copyright (C) 2015 Ludwig Knüpfer <ludwig.knuepfer@fu-berlin.de>,
+ *                    Martine Lenders <mlenders@inf.fu-berlin.de>
+ *                    Kaspar Schleiser <kaspar@schleiser.de>
+ *                    Ell-i open source co-operative
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser General
+ * Public License v2.1. See the file LICENSE in the top level directory for
+ * more details.
+ */
+
+#include <assert.h>
+#include <err.h>
+#include <errno.h>
+#include <fcntl.h>
+#include <inttypes.h>
+#include <stdio.h>
+#include <stdint.h>
+#include <stdlib.h>
+#include <string.h>
+#include <sys/ioctl.h>
+#include <unistd.h>
+
+/* needs to be included before native's declarations of ntohl etc. */
+#include "byteorder.h"
+
+#ifdef __MACH__
+#define _POSIX_C_SOURCE
+#include <net/if.h>
+#undef _POSIX_C_SOURCE
+#include <sys/types.h>
+#include <ifaddrs.h>
+#include <net/if_dl.h>
+#elif defined(__FreeBSD__)
+#include <sys/socket.h>
+#include <net/if.h>
+#include <ifaddrs.h>
+#include <net/if_dl.h>
+#else
+#include <net/if.h>
+#include <linux/if_tun.h>
+#include <linux/if_ether.h>
+#include <linux/if_packet.h>
+#endif
+
+#include "native_internal.h"
+
+#include "net/netdev.h"
+#include "netdev_raw802154.h"
+#include "net/netopt.h"
+
+#include "periph/hwrng.h"
+
+#include "async_read.h"
+
+#define ENABLE_DEBUG (0)
+#include "debug.h"
+
+#ifndef ETH_P_IEEE802154
+#define ETH_P_IEEE802154 0x00F6
+#endif
+
+#define RAW802154_DEFAULT_ADDR_PAN      (0x0023)
```

Comment:
With https://github.com/RIOT-OS/RIOT/pull/10384 close to being merged, this is not required anymore.